### PR TITLE
implement `@time` and `@timev` in terms of `@timed`

### LINKED ISSUE #47056: `@timed` does not report compile/recompile times whereas `@time` does.
I'm interested in collecting compile/recompile time of the expressions I'm evaluating because they happen to be large and automatically generated.

I can read compilation time from my screen when using `@time`, but when I attempt to collect this information as data for later use with `@timed` instead, the compilation/recompilation times seem not to appear within the resulting `NamedTuple`.

As a workaround, I'm using `Base.cumulative_compile_tim*` functions because I've found in the sources that this is what `@time` is using under the hood. But they are not documented and I'm unsure how or whether I'm supposed to do this.

Can the compilation/recompilation times be added to the output of `@timed`?  
As an alternative, will the `Base.cumulative_compile_tim*` functions be eventually documented and/or exported?

### PR DESCRIPTION
**Pull request by ericphanson**
closes https://github.com/JuliaLang/julia/issues/47056

I suspect there was some reason this wasn't done in the first place, but I figured opening a PR could be a way to discuss that.

~~Needs tests still for the new `@timed` fields.~~

### CODE REVIEW

**Comment by IanButterworth** [comment]
Sounds good to me. We'd need to check that we're not hitting unexpected compilation due to this change.

**Comment by ericphanson** [comment]
Do you know a good way to do that?

I tried the example from https://github.com/JuliaLang/julia/issues/48024#issue-1512995272:

```julia
❯ ./julia --startup-file=no -q
julia> double(x::Real) = 2x
double (generic function with 1 method)

julia> calldouble(container) = double(container[1])
calldouble (generic function with 1 method)

julia> calldouble2(container) = calldouble(container)
calldouble2 (generic function with 1 method)

julia> @time @eval calldouble([1.0])
  0.009445 seconds (2.48 k allocations: 116.594 KiB, 30.16% compilation time) # master
  0.007254 seconds (2.48 k allocations: 116.594 KiB, 30.60% compilation time) # PR
2.0

julia> @time @eval calldouble2(1.0)
  0.002668 seconds (3.81 k allocations: 194.422 KiB, 93.59% compilation time) # master
  0.003196 seconds (3.81 k allocations: 194.422 KiB, 95.07% compilation time) # PR
2.0
```
which at least looks similar, though I don't know if it means anything.

File: base/timing.jl [hunk 1]
```diff
@@ -500,19 +480,34 @@ julia> propertynames(stats.gcstats)
...
+julia> stats.recompile_time
+0.0
+
 ```
 
 !!! compat "Julia 1.5"
     The return type of this macro was changed from `Tuple` to `NamedTuple` in Julia 1.5.
+
+!!! compat "Julia 1.11"
+    The compilation and recompilation time fields were added in Julia 1.9.
```
Comment (ericphanson) [inline base/timing.jl:496]:
```suggestion
    The compilation and recompilation time fields were added in Julia 1.11.
```

**Comment by IanButterworth** [comment]
Needs news for the new fields

**Comment by ericphanson** [comment]
do we like these field names? we could do `compilation_time` instead of `compile_time` since that's how I refer to it in the docstring, though having it shorter seems a bit nicer

**Comment by IanButterworth** [comment]
`compile_time` is understandable and shorter is nicer IMO

File: base/timing.jl [hunk 2]
```diff
@@ -500,19 +480,34 @@ julia> propertynames(stats.gcstats)
...
+julia> stats.recompile_time
+0.0
+
 ```
 
 !!! compat "Julia 1.5"
     The return type of this macro was changed from `Tuple` to `NamedTuple` in Julia 1.5.
+
+!!! compat "Julia 1.11"
+    The compilation and recompilation time fields were added in Julia 1.11.
```
Comment (inkydragon) [inline base/timing.jl:496]:
```suggestion
    The `compile_time` and `recompile_time` fields were added in Julia 1.11.
```

**Review by IanButterworth — APPROVED** [review]
LGTM. Avoids duplication and the consistency in outputs seems good to me

**Pull request merged**